### PR TITLE
[Security] do not validate passwords when the hash is null

### LINKED ISSUE #34775: [Security] isPasswordValid() throws an exception when the user has no password set
**Symfony version(s) affected**: 4.4.0

**Description**  
I'm getting an exception thrown when a user who has no password set at all (their password field is null, they use another method to log in) attempts to change their password:

paveikslas

I'm not sure if this will be considered a bug, but I don't think that passwordEncoders should assume that the value stored as the password will always be a string.

From looking at the code, it seems that `needsRehash()` is perhaps the only method which makes that assumption. I was about to suggest to allow non-string values in `needsRehash()`, but in this particular case, `needsRehash()` is called as part of checking the supplied password for validity: 
https://github.com/symfony/symfony/blob/6c163906f2ca4e3498061380d8f6b40e374badd2/src/Symfony/Component/Security/Core/Encoder/MigratingPasswordEncoder.php#L51

Presumably, this is to avoid extra checks with other encoders if the password hash appears valid to the current encoder. This doesn't seem necessary, so perhaps just removing this check would be acceptable?

Alternatively, perhaps `$encoded` could be cast to string in that call?

**How to reproduce**  
1. Create a user without a password.
2. Try logging in with that user with a password.

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #34775
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Security/Core/Encoder/UserPasswordEncoder.php [hunk 1]
```diff
@@ -44,6 +44,10 @@ public function isPasswordValid(UserInterface $user, $raw)
     {
         $encoder = $this->encoderFactory->getEncoder($user);
 
+        if (null === $user->getPassword()) {
```
Comment (rimas-kudelis) [inline src/Symfony/Component/Security/Core/Encoder/UserPasswordEncoder.php]: This block could just go above the $encoder assignment.

**Comment by nicolas-grekas** [comment]
CI is broken :)

**Comment by xabbuh** [comment]
CI is green

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @xabbuh.

**Pull request merged**